src/Board.cpp

```diff
@@ -0,0 +1,182 @@
+#pragma once
+#include "board.h"
+ /*Constructor for the Board class.
+ Takes a string representing the initial board state and initializes the board with the pieces.
+
+ Parameters:
+ - initBoard: A string representing the initial board state. Each character in the string represents
+   a piece on the board (uppercase for white, lowercase for black), or a space for an empty square.
+   The string should be ordered from the top-left corner (A8) to the bottom-right corner (H1).*/
+Board::Board(const string& initBoard)
+    :m_turn(PieceColor::WHITE)
+{
+    for (size_t i = 0; i < initBoard.length(); ++i) {
+        auto piece = PieceFactory::createPiece(initBoard[i], convertToLocation(i));
+        if (piece) m_board.push_back(std::move(piece));
+    }
+}
+
+/*Destructor for the Board class.
+ Cleans up resources associated with the board.*/ 
+Board::~Board() {}
```

Comment:
If there's no special implementation for the destructor then you don't need to explicitly add it. The compiler will do that for you